### PR TITLE
Update engine to use include paths

### PR DESCRIPTION
**Pull request by jacooobi**
As listed in [spec](https://github.com/codeclimate/spec/blob/master/SPEC.md#input) `excluded_paths` is now deprecated.

### CODE REVIEW

File: lib/cc/engine/rubymotion.rb [hunk 1]
```diff
@@ -42,6 +40,29 @@ def run
...
+          end
+        end.flatten.compact
+      end
+
+      def build_files_with_exclusions(exclusions)
+        files = Dir.glob("**/*.rb")
+        files.reject { |f| exclusions.include?(f) }
+      end
+
       def exclude?(path)
```
Comment (gdiggs) [inline lib/cc/engine/rubymotion.rb]: This method is now unused, right?

File: lib/cc/engine/rubymotion.rb [hunk 2]
```diff
@@ -42,6 +40,29 @@ def run
 
       private
 
+      def files_to_analyze
```
Comment (gdiggs) [inline lib/cc/engine/rubymotion.rb:44]: What do you think of pulling these methods out into their own class? Something that could call like `AnalyzableFiles.new(@engine_config).files`. I think that would keep this `Rubymotion` class a bit simpler and closer to following SRP

File: Gemfile.lock [hunk 3]
```diff
@@ -42,20 +42,13 @@ GEM
       method_source (~> 0.8.1)
       slop (~> 3.4)
     rack (1.6.0)
-    rack-protection (1.5.3)
```
Comment (gdiggs) [inline Gemfile.lock:45]: What caused these Gemfile.lock changes? They seem good but I'm surprised there isn't a corresponding `Gemfile` change

Comment (jacooobi) [inline Gemfile.lock:45]: To be honest I just ran bundle install. The changes seemed good ( like you said ) so I decided to push it.

File: lib/cc/engine/rubymotion.rb (hunk 1 above)
Comment (jacooobi) [inline lib/cc/engine/rubymotion.rb]: Good catch!

File: lib/cc/engine/rubymotion.rb (hunk 2 above)
Comment (jacooobi) [inline lib/cc/engine/rubymotion.rb:44]: It's a good separation of concerns and I can do this. However, I think if we decide to go with a class approach, we should keep it consistent and do the same in every other engine. That would require small changes to engines which already have switched from `exclude_paths` to `include_paths` such as Rubocop, Eslint, and CSSlint.

**Comment by jacooobi** [comment]
@GordonDiggs I've followed your sugestion and moved the analyzable files logic into a separate class. Mind taking a look?

File: Gemfile.lock (hunk 3 above)
Comment (gdiggs) [inline Gemfile.lock:45]: Can we remove the changes for now? At the very least they should be in a separate PR if they're not related to this functionality.
